docs/extend/extend-apiml/api-mediation-security.md

```diff
@@ -594,3 +648,29 @@ in following shell scripts:
 - `$ZOWE_RUNTIME/api-mediation/scripts/api-mediation-start-catalog.sh`
 - `$ZOWE_RUNTIME/api-mediation/scripts/api-mediation-start-discovery.sh`
 - `$ZOWE_RUNTIME/api-mediation/scripts/api-mediation-start-gateway.sh`
+
+## Security Service Client library
+
+The `security-service-client-spring` library enables authentication and protection using security providers.
+The library contains providers, filters and handlers as Spring components. The `security-service-client-spring` library enables any Spring client to authenticate mainframe user credentials.
+
+ The following providers process authentication requests:
+  
+   - `com.ca.apiml.security.login.GatewayLoginProvider` class that verifies mainframe credentials 
+   - `com.ca.apiml.security.token.GatewayTokenProvider` class that authenticates the JWT token 
+
+The library also contains the following Spring Security filters:
+
+- The `com.ca.apiml.security.content.BasicContentFilter` to authenticate the credentials from the basic authorization header. The `com.ca.apiml.security.content.BasicContentFilter` can be used in `SecurityConfiguration` to secure content with basic authentication. Also the filter can be used in the `SecurityConfiguration` class (see a [sample](https://github.com/zowe/api-layer/blob/master/api-catalog-services/src/main/java/com/ca/mfaas/apicatalog/security/SecurityConfiguration.java)) to process the `/login` requests.
+- The `com.ca.apiml.security.content.CookieContentFilter` to authenticate the JWT token that is stored in the cookie by extracting the JWT token from it. This filter can be used in a `SecurityConfiguration` in order to secure content with a token stored in the cookie. The token is extracted from the cookie and passed to the `GatewayTokenProvider`, which calls the `/query` endpoint.
+- The `com.ca.apiml.security.login.LoginFilter` to process authentication requests with a username and password in a JSON format. This filter can be used in a `SecurityConfiguration` class to process the `/login` request.
```

Comment:
This needs a better description. I am not even sure if it is really useful for other services. This 'creates' login endpoint. The main purpose is to use it in GW. However, Catalog needs to replicate the same endpoint so it was moved into the library. Catalog backend needs it for UI which is calling it.